### DISCUSSION TITLE
Liquidsoap switch only working a few days

### DISCUSSION THREAD

**Discussion by paulrusu** [savonet/liquidsoap#1773] [Q&A]
I have the following code. It works fine but only the first 2 or 3 days. The 4 or 5th time it has to play a file from devotional/poezii playlist it doesn't anymore.

I don't have errors in the liquidsoap log. I see in the log it simply doesn't prepare and ffmpeg accept songs from the 2 playlists anymore only the first 4-5 times. Only the music which is played in the rest continues.

    music1 = playlist("~/music/music1", mode="random")
    music2 = playlist("~/music/music2", mode="random")
    music4 = playlist("~/music/music4", mode="random")
    security = single("~/music/default.mp3")
    radio = rotate([music1, music2, music4])
    radio = fallback(track_sensitive = false,
      [fallback([request.queue(id="request"),
      radio]),security])

    devotionale = playlist("~/devotionale")
    poezii = playlist("~/poezii")

    radio = switch(track_sensitive=false, [ ({ 8h30m-8h59m or 20h30m-20h59m }, once(devotionale)),
                                    ({ 9h00m-9h29m or 21h00m-21h29m }, once(poezii)),
                                    ({true}, radio) ])

**Comment by smimram** [comment]
The `once` operator says that you should play one track of `devotionale` / `poezii`, which is probably not what you want here?

**Comment by paulrusu** [comment]
I want to play one track at that time each day. If i don't do once it will play many songs in that interval. I just want one and then normal music can resume.

**Reply by spy-reality** [reply]
Did you find a solution?

**Comment by martinkirch** [comment]
I you want to play one track from a source _each day_, I suggest you replace `once` (which plays one time ever) by `delay(3600.*24., devotionale)` : after playing a track, it will make `devotionale` artificailly unavailable for `3600.*24.` seconds (one day). Depending on track length, you might prefer a bit shorter delay.
